src/MediaHelper.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\collabora_online;
+
+use Drupal\file\Entity\File;
+use Drupal\media\Entity\Media;
+
+/**
+ * Service to assist with media entities.
+ */
+class MediaHelper {
+
+  /**
+   * Gets the file referenced by a media entity.
+   *
+   * @param \Drupal\media\Entity\Media $media
+   *   The media entity.
+   *
+   * @return \Drupal\file\FileInterface|null
+   *   The file entity, or NULL if not found.
+   */
+  public function getFileForMedia(Media $media) {
+    $fid = $media->getSource()->getSourceFieldValue($media);
+    $file = File::load($fid);
+
+    return $file;
+  }
+
+  /**
+   * Gets a file based on the media id.
+   *
+   * @param int|string $id
+   *   Media id which might be in strong form like '123'.
+   *
+   * @return \Drupal\file\FileInterface|null
+   *   File referenced by the media entity, or NULL if not found.
+   */
+  public function getFileForMediaId($id) {
+    /** @var \Drupal\media\MediaInterface|null $media */
+    $media = \Drupal::entityTypeManager()->getStorage('media')->load($id);
+    return $this->getFileForMedia($media);
+  }
+
+  /**
+   * Sets the file entity reference for a media entity.
+   *
+   * @param \Drupal\media\Entity\Media $media
+   *   Media entity to be modified.
+   * @param \Drupal\file\Entity\File $source
+   *   File entity to reference.
+   */
+  public function setMediaSource(Media $media, File $source) {
+    $name = $media->getSource()->getSourceFieldDefinition($media->bundle->entity)->getName();
```

Comment:
Btw @AaronGilMartinez This one is also dirty, but I don't think we can easily avoid it.

In `MediaType::sourcePluginCollection()` there is a `Drupal::service()` call:
```php
    if (!$this->sourcePluginCollection && $this->source) {
      $this->sourcePluginCollection = new DefaultSingleLazyPluginCollection(\Drupal::service('plugin.manager.media.source'), $this->source, $this->source_configuration);
    }
    return $this->sourcePluginCollection;
```

Reply:
Indeed, not sure if we can avoid.